sdks/java/core/src/main/java/org/apache/beam/sdk/coders/BeamRecordCoder.java

```diff
@@ -56,7 +56,7 @@ public BeamRecordType getRecordType() {
   @Override
   public void encode(BeamRecord value, OutputStream outStream)
       throws CoderException, IOException {
-    nullListCoder.encode(value.getNullFields(), outStream);
+    nullListCoder.encode(scanNullFields(value), outStream);
```

Comment:
many `Coder`s cannot encode a null value, like `BigDecimalCoder`, `BigEndianIntegerCoder`, `BigEndianLongCoder`, `StringUtf8Coder` and so on.

Here I handle null values separately, another choice is using NullableCoder to wrap each `Coder`. Is the latter one preferred ?